File: app/src/main/java/io/vinter/trackmyanime/network/service/UserService.java
package io.vinter.trackmyanime.network.service;

import io.reactivex.Single;
import io.vinter.trackmyanime.entity.user.LoginResponse;
import io.vinter.trackmyanime.network.form.LoginForm;
import retrofit2.Response;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface UserService {
    @POST("/user/login")
    Single<LoginResponse> getToken(@Body LoginForm form);

    @POST("/user/signup")
    Single<LoginResponse> register(@Body LoginForm form);
}

File: app/src/main/java/io/vinter/trackmyanime/network/service/UserService.kt
package io.vinter.trackmyanime.network.service

import io.reactivex.Single
import io.vinter.trackmyanime.entity.user.LoginResponse
import io.vinter.trackmyanime.network.form.LoginForm
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST

interface UserService {
    @POST("/user/login")
    fun getToken(@Body form: LoginForm): Single<LoginResponse>

    @POST("/user/signup")
    fun register(@Body form: LoginForm): Single<LoginResponse>
}
